### ISSUE TITLE
[PyTorch] Segfault when compiling a mlir made by export

### ISSUE THREAD

**Issue by meisonoda-ux** [iree-org/iree#24448]
### What happened?

When I tried to compile ResNet50 nn model, export seems to work fine and produce a mlir that looks roughly correct but when i try to compile said mlir, i get a segfault. I've observed that the smallest nn model i get a segfault on is 
`
class MiscNet10(nn.Module):
    def __init__(self, num_classes=10):
        super(MiscNet10, self).__init__()
        
        # Layer 1: Standard Convolution (Entry point)
        self.conv1 = nn.Conv2d(3, 8, kernel_size=3, stride=1, padding=1)
        self.bn1 = nn.BatchNorm2d(8)
        self.relu = nn.ReLU()

        # Layer 9: Global Average Pooling (The bridge to Linear)
        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
        
        # Layer 10: Fully Connected (The Head)
        self.fc = nn.Linear(8, num_classes)

    def forward(self, x):
        # Apply the layers
        x = self.relu(self.bn1(self.conv1(x)))
        
        x = self.avgpool(x)
        x = torch.flatten(x, 1)
        x = self.fc(x)
        return x`

However, the segfault disappears if i delete everything concerning BatchNorm2d. The segfault free version looks like this:

`
class MiscNet10(nn.Module):
    def __init__(self, num_classes=10):
        super(MiscNet10, self).__init__()
        
        # Layer 1: Standard Convolution (Entry point)
        self.conv1 = nn.Conv2d(3, 8, kernel_size=3, stride=1, padding=1)
        self.relu = nn.ReLU()

        # Layer 9: Global Average Pooling (The bridge to Linear)
        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
        
        # Layer 10: Fully Connected (The Head)
        self.fc = nn.Linear(8, num_classes)

    def forward(self, x):
        # Apply the layers
        x = self.relu(self.conv1(x))
        
        x = self.avgpool(x)
        x = torch.flatten(x, 1)
        x = self.fc(x)
        return x`

### Steps to reproduce your issue

1. Run the following code using the following test script:
`def test_misc10():
    model = MiscNet10()
    example_input = torch.randn(1, 3, 224, 224)

    print("--- Stage 1: Capturing Graph (Torch -> MLIR) ---")
    export_output = aot.export(model, example_input, strict_export=True)
    
    export_output.save_mlir("misc10.mlir")
    print("Saved misc10.mlir")

    print("\n--- Stage 2: Compiling (MLIR -> SPIR-V) ---")
    try:
        binary = export_output.compile(
            save_to="misc10.vmfb",
            target_backends=["vulkan-spirv"]
        )
        print("Success! binary saved to misc10.vmfb")
    except Exception as e:
        print(f"Compilation Failed: {e}")`
2. You should see a segfault occur

### Version information

v3.11.0
